app/models/user.rb

```diff
@@ -107,6 +107,11 @@ def inactive_message
     "Sorry, this account has been disabled."
   end
 
+  # Returns all users who match the query.
+  def self.search_from_query(members, query)
+    User.enabled.where.not(id: members).where(arel_table[:username].matches(query)).pluck(:username)
```

Comment:
btw, `search_from_query` plucks the username, wouldn't have guessed that by the name. Would be cool if it would return the ActiveRelation Object (just remove `.pluck(:username)` in the end)